lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -223,9 +264,44 @@ def create_instance(module, client, instance_name):
             resp = resp['operations'][0]
         except botocore.exceptions.ClientError as e:
             module.fail_json(msg='Unable to create instance {0}, error: {1}'.format(instance_name, e))
-        changed = True
 
-    inst = _find_instance_info(client, instance_name)
+        inst = _find_instance_info(client, instance_name)
+
+        # Wait for instance to become running
+        if wait:
+            while (wait_max > time.time()) and (inst is not None and inst['state']['name'] != "running"):
+                try:
+                    time.sleep(2)
+                    inst = _find_instance_info(client, instance_name)
+                except botocore.exceptions.ClientError as e:
```

Comment:
It looks like if it isn't a permission error or request expired the exceptions are ignored. It would probably be better just to retry on a specific exception or two and otherwise call fail_json than mask undisclosed exceptions.